TestUtilities/Mocks/NetworkInstrumentationMocks.swift

```diff
@@ -28,15 +47,37 @@ public class RelativeTracingUUIDGenerator: TraceIDGenerator {
     }
 
     public func generate() -> TraceID {
+        return queue.sync {
+            defer { uuid = uuid + (0, count) }
+            return uuid
+        }
+    }
+}
+
+public class RelativeSpanIDGenerator: SpanIDGenerator {
+    private(set) var uuid: SpanID
+    internal let count: UInt64
+    private let queue = DispatchQueue(label: "queue-RelativeTracingUUIDGenerator-\(UUID().uuidString)")
```

Comment:
suggestion/ Using `@ReadWriteLock var uuid: SpanID` might be much cleaner than syncing with queue. We can update `RelativeTracingUUIDGenerator` with this change as well. Rule of thumb: with property wrappers being available, we can simplify a lot of synchronisation in tests by replacing dummy queues with this convenient lock.